casper/src/test/scala/coop/rchain/casper/MergeabilityRules.scala

```diff
@@ -563,32 +563,31 @@ trait MergeabilityRules {
     "(P!) P!"   -> PeekedNoMatch(P_)(P0, S0)(S1),
     "4! (P!)"   -> PeekedNoMatch(F_)(P1, S1)(S0),
     // The case below can merge iff peek block is executed as first - conflicts for now
-    "4! P!"     -> CouldMatchSameConflicts(F_)(P_)(S0),
-    "4! P!"     -> CouldMatchSameMerges(F0)(P1)(S0, S1),
-    "4! PX"     -> SamePolarityMerge(F_)(P1)(S0),
-    "4X PX"     -> SamePolarityMerge(F_)(P_)(Nil),
-    "P! (C!)"   -> PersistentNoMatch(P_)(C0, S0)(S1),
-    "P! (P!)"   -> PeekedNoMatch(P_)(P1, S1)(S0),
-    "P! C!"     -> CouldMatchSameConflicts(P_)(C_)(S0),
-    "P! P!"     -> CouldMatchSameMerges(P0)(P1)(S0, S1),
-    "PX (C!)"   -> PersistentNoMatch(P_)(C_, S0)(Nil),
-    "PX (P!)"   -> PeekedCouldMatch(P_)(P_, S0)(Nil),
-    "PX C!"     -> SamePolarityMerge(P0)(C_)(S1),
-    "PX CX"     -> SamePolarityMerge(P_)(C_)(Nil),
-    "PX PX"     -> SamePolarityMerge(P_)(P_)(Nil),
-    "!C P!"     -> HadItsMatch(S0)(P_)(C0, S1),
-    "(!4) (P!)" -> VolatileEvent(S0, F_)(P_, S0)(Nil),
-    "!C (P!)"   -> PeekedNoMatch(S0)(P_, S1)(C0),
-    "(P!) (C!)" -> PeekedCouldMatch(P_, S0)(C_, S0)(Nil),
-    "(!C) (P!)" -> PeekedCouldMatch(S0, C_)(P_, S0)(Nil),
-    "(P!) C!"   -> PeekedCouldMatch(C_)(P0, S0)(S1),
-    "(P!) CX"   -> PeekedCouldMatch(C_)(P_, S0)(Nil),
-    "4X (P!)"   -> PeekedCouldMatch(F_)(P_, S0)(Nil),
-    "P! CX"     -> SamePolarityMerge(P_)(C1)(S0),
-    "4X P!"     -> SamePolarityMerge(F0)(P_)(S1),
-    "PX P!"     -> SamePolarityMerge(P0)(P_)(S1),
-    // TODO:
-    "P! P!"       -> CouldMatchSameConflicts(P_)(P_)(S0), // conflicts until replay errors are fixed
+    "4! P!"       -> CouldMatchSameConflicts(F_)(P_)(S0),
```

Comment:
Is there a way to derive the right-hand side from the left instead of a long list of cases?

Reply:
There is no simple answer to this question. Some cases from the left may have more than one representation on the right, depending on whether we want them to conflict or not. I plan to add auto derivation in a separate PR